### ISSUE TITLE
FN S2857: No issue is raised if the SQL string is assembled with constant interpolated strings

### ISSUE THREAD

**Issue by mary-georgiou-sonarsource** [SonarSource/sonar-dotnet#5947]
### Description

S2857 does not raise an issue if the SQL string is assembled with constant interpolated strings

### Repro steps
```csharp
    class Examples
    {
        public void VariousSqlKeywords()
        {
            const string s1 = "TRUNCATE";
            const string s2 = "TABLE HumanResources.JobCandidate;";
            const string noncompliant = $"{s1}{s2}"; // FN
        }
    }
```

### Expected behavior

Raises an issue

### Actual behavior

Does not raise an issue.

### Known workarounds

None

### Related information
* C#/VB.NET Plugins version: 8.42.0.51121

**Comment by mary-georgiou-sonarsource** [comment]
Duplicate of https://github.com/SonarSource/sonar-dotnet/issues/5942
